app/src/main/java/edu/uc/groupproject/covid19tracker/BarChartFragment.kt

```diff
@@ -0,0 +1,171 @@
+        // while interacting with the UI.
+        dummyButton?.setOnTouchListener(delayHideTouchListener)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        activity?.window?.addFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS)
+
+        // Trigger the initial hide() shortly after the activity has been
+        // created, to briefly hint to the user that UI controls
+        // are available.
+        delayedHide(100)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        activity?.window?.clearFlags(WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS)
+
+        // Clear the systemUiVisibility flag
+        activity?.window?.decorView?.systemUiVisibility = 0
+        show()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        dummyButton = null
+        fullscreenContent = null
+        fullscreenContentControls = null
+    }
+
+    private fun toggle() {
+        if (visible) {
+            hide()
+        } else {
+            show()
+        }
+    }
+
+    private fun hide() {
+        // Hide UI first
+        fullscreenContentControls?.visibility = View.GONE
+        visible = false
+
+        // Schedule a runnable to remove the status and navigation bar after a delay
+        hideHandler.removeCallbacks(showPart2Runnable)
+        hideHandler.postDelayed(hidePart2Runnable, UI_ANIMATION_DELAY.toLong())
+    }
+
+    @Suppress("InlinedApi")
+    private fun show() {
+        // Show the system bar
+        fullscreenContent?.systemUiVisibility =
+            View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN or
+                    View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+        visible = true
+
+        // Schedule a runnable to display UI elements after a delay
+        hideHandler.removeCallbacks(hidePart2Runnable)
+        hideHandler.postDelayed(showPart2Runnable, UI_ANIMATION_DELAY.toLong())
+        (activity as? AppCompatActivity)?.supportActionBar?.show()
+    }
+
+    /**
+     * Schedules a call to hide() in [delayMillis], canceling any
+     * previously scheduled calls.
+     */
+    private fun delayedHide(delayMillis: Int) {
+        hideHandler.removeCallbacks(hideRunnable)
+        hideHandler.postDelayed(hideRunnable, delayMillis.toLong())
+    }
+
+    companion object {
+        /**
+         * Whether or not the system UI should be auto-hidden after
+         * [AUTO_HIDE_DELAY_MILLIS] milliseconds.
+         */
+        private const val AUTO_HIDE = true
+
+        /**
+         * If [AUTO_HIDE] is set, the number of milliseconds to wait after
+         * user interaction before hiding the system UI.
+         */
+        private const val AUTO_HIDE_DELAY_MILLIS = 3000
+
+        /**
+         * Some older devices needs a small delay between UI widget updates
+         * and a change of the status and navigation bar.
+         */
+        private const val UI_ANIMATION_DELAY = 300
```

Comment:
Added new fragment to show bar chart on a larger screen when clicked on

Reply:
Ah, cool!